app/OpenGraphImage.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace App;
+
+use GDText\Box;
+use GDText\Color;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Str;
+
+class OpenGraphImage
+{
+    const PADDING = 50;
+    const GUTTER = 15;
+
+    protected $fileName;
+
+    /**
+     * Generate an open graph image file name.
+     *
+     * @param string $uniqueId A unique key to identify the image by.
+     * @param string $name Name of the file.
+     */
+    public static function makeFileName($uniqueId, $name): string
+    {
+        return "{$uniqueId}_" . Str::slug($name, '-') . '.png';
+    }
+
+    public function __construct($fileName)
+    {
+        $this->fileName = $fileName;
+    }
+
+    public function storageDirectory(): string
+    {
+        return config('opengraph.image_directory_name') . '/';
+    }
+
+    public function storagePath(): string
+    {
+        return Storage::disk('public')->path('') . $this->storageDirectory() . "{$this->fileName}";
+    }
+
+    public function makeStorageDirectory(): bool
+    {
+        if (Storage::exists($this->storageDirectory())) {
+            return false;
+        }
+
+        Storage::makeDirectory($this->storageDirectory());
+        return true;
+    }
+
+    public function removeDuplicates()
```

Comment:
"duplicates" is a little misleading in this method name, since neither the filenames are duplicate, nor the image contents (if the package name is different). Maybe something like `deleteImagesForProject`? That's super cumbersome, but more accurate. What do you think?

Reply:
Agree. Thank you! I struggled naming this one.